src/citrine/informatics/predictors/graph_predictor.py

```diff
@@ -35,11 +35,9 @@ class GraphPredictor(Serializable['GraphPredictor'], Predictor):
 
     predictors = _properties.List(_properties.Union(
         [_properties.UUID, _properties.Object(Predictor)]), 'config.predictors')
-    training_data = _properties.List(_properties.Object(DataSource), 'config.training_data')
+    training_data = _properties.List(
+        _properties.Object(DataSource), 'config.training_data', default=[])
```

Comment:
The backend has a `None` default, so this turns it back into an empty list.

Reply:
Using a mutable default value could lead to strange behavior. Is there a way to make the default immutable (maybe a using tuple but I'm not sure how this would affect our typed serialization) or translate `None` to an empty list in the constructor?

Reply:
But this is on the class level. Even translating it inside the constructor would lead to the same list getting used for every deserialization.

Reply:
I think we'll need to translate `None` to a `[]` at some point during deserialization, but looking at the code I'm not sure where the best place to do that is.

Reply:
The values are read into a new list each time an object is deserialized here: https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/_serialization/properties.py#L370

So this shouldn't be an issue. Thanks to @maxhutch for adding a test to check.